content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md

```diff
@@ -0,0 +1,317 @@
+
+        app: polling-app
+
+    spec: 
+
+      selector: 
+
+        matchLabels:                    # This deployment applies to the Pods matching the specified labels 
+
+          app: polling-app
+
+          tier: mysql
+
+      strategy: 
+
+        type: Recreate
+
+      template:                         # Template for the Pods in this deployment 
+
+        metadata: 
+
+          labels:                       # Labels to be applied to the Pods in this deployment 
+
+            app: polling-app
+
+            tier: mysql
+
+        spec:                           # The spec for the containers that will be run inside the Pods in this deployment 
+
+          containers: 
+
+          - image: mysql:5.6            # The container image 
+
+            name: mysql
+
+            env:                        # Environment variables passed to the container  
+
+            - name: MYSQL_ROOT_PASSWORD 
+
+              valueFrom:                # Read environment variables from kubernetes secrets 
+
+                secretKeyRef: 
+
+                  name: mysql-root-pass
+
+                  key: password
+
+            - name: MYSQL_DATABASE
+
+              valueFrom: 
+
+                secretKeyRef: 
+
+                  name: mysql-db-url
+
+                  key: database
+
+            - name: MYSQL_USER
+
+              valueFrom: 
+
+                secretKeyRef: 
+
+                  name: mysql-user-pass
+
+                  key: username
+
+            - name: MYSQL_PASSWORD
+
+              valueFrom: 
+
+                secretKeyRef: 
+
+                  name: mysql-user-pass
+
+                  key: password
+
+            ports: 
+
+            - containerPort: 3306        # The port that the container exposes        
+
+              name: mysql
+
+            volumeMounts: 
+
+            - name: mysql-persistent-storage  # This name should match the name specified in `volumes.name` 
+
+              mountPath: /var/lib/mysql
+
+          volumes:                       # A PersistentVolume is mounted as a volume to the Pod   
+
+          - name: mysql-persistent-storage
+
+            persistentVolumeClaim: 
+
+              claimName: mysql-pv-claim
+
+In short, we are creating the following four resources in the above
+manifest file: 
+
+-   PV
+-   A PVC for requesting access to the PV resource
+-   A service for having a static endpoint for the MySQL database
+-   A deployment for running and managing the MySQL Pod
+
+The MySQL container reads database credentials from environment
+variables. The environment variables access these credentials from
+Kubernetes secrets.\
+ For the next step, you can start a Minikube cluster to create
+Kubernetes secrets to store database credentials and deploy the MySQL
+instance.
+
+### Summary
+
+In this article, you learn about how to deploy MySQL on
+[Kubernetes](https://myhsts.org/tutorial-review-of-17-essential-topics-for-mastering-kubernetes.php)
```

Comment:
Omit this hyperlink. The blog already links to the top of the site.

Reply:
I removed it